pcdscalc/be_lens_calcs.py

```diff
@@ -925,15 +942,409 @@ def find_z_pos(energy, lens_set, spot_size_fwhm, material=None,
     waist = lam / np.pi * focal_length / w_unfocused
     rayleigh_range = np.pi * waist ** 2 / lam
 
-    logger.info("waist          : %.3e" % waist)
-    logger.info("waist FWHM     : %.3e" % (waist
-                                           * FWHM_SIGMA_CONVERSION / 2.0))
-    logger.info("rayleigh_range : %.3e" % rayleigh_range)
-    logger.info("focal length   : %.3e" % focal_length)
+    logger.info("waist          : %.3e", waist)
+    logger.info("waist FWHM     : %.3e", waist * FWHM_SIGMA_CONVERSION / 2.0)
+    logger.info("rayleigh_range : %.3e", rayleigh_range)
+    logger.info("focal length   : %.3e", focal_length)
 
     w = gaussian_fwhm_to_sigma(spot_size_fwhm) * 2
     delta_z = rayleigh_range * np.sqrt((w / waist) ** 2 - 1)
     z1 = focal_length - delta_z
     z2 = focal_length + delta_z
     z_position = (z1, z2)
     return z_position
+
+
+def plan_set(energy, z_offset, z_range, beam_size_unfocused, size_horizontal,
+             size_vertical=None, exclude=[], max_tot_number_of_lenses=25,
+             max_each=5, focus_before_sample=False):
+    """
+    Macro to help plan for what lens set to use.
+
+    Parameters
+    ----------
+    energy : number
```

Comment:
Most of the input parameters here need to have units associated with them. It's probably good to check other functions as well so that we don't get confused later.

Reply:
Ok, i added to some of these functions in 279f107 but not to all of them (because I am not 100 % sure)
I am almost sure that the `radius` that we're using from the lens_set and in most of the functions are in meters, so i can add that too but I am not 100 % convinced.

Reply:
Ok we need to figure these out then, if the wrong units are used then the calcs will not work as expected

Reply:
Yeah I agree - i'm going to go back and try to figure out all of them. Based on the old `curren_set` file we have these: 
```
[(3, 0.0001, 1, 0.0002),
 (1, 0.0001, 1, 0.0003, 1, 0.0005),
 (2, 0.0001, 1, 0.0005)]
```
So it make sense that they would be in meters, because we have the `LENS_RADII = [50e-6, 100e-6, 200e-6, 300e-6,
              500e-6, 1000e-6, 1500e-6, 2000e-6, 3000e-6]` which are in `um`